src/main/java/duke/parser/Brain.java

```diff
@@ -0,0 +1,203 @@
+     */
+    public void createDeadline(String input, DataStore dataStore, Ui userInt) {
+        input = input.split(" ", 2)[1].strip();
+        String query = input.split("/by")[0];
+        String limit = input.split("/by")[1];
+        String procLimit = processDate(limit);
+
+        Deadline deadlineTask = new Deadline(query, procLimit);
+        dataStore.add(deadlineTask);
+
+        userInt.say("Got it. I've added this task: ");
+        userInt.say("\t" + deadlineTask.toString());
+        userInt.say("Now you have " + dataStore.length() + " tasks in the list.");        
+    }
+
+    /**
+     *
+     * Creates an Event task given a description and timing.
+     *
+     * @param input the user's input with the description and timing.
+     * @param dataStore the list containing the up-to-date task records.
+     * @param userInt the user interface object for logging.
+     */
+    public void createEvent(String input, DataStore dataStore, Ui userInt) {
+        input = input.split(" ", 2)[1].strip();
+        String query = input.split("/at")[0];
+        String datetime = input.split("/at")[1];
+        String procDate = processDate(datetime);
+
+        Event eventTask = new Event(query, procDate);
+        dataStore.add(eventTask);
+
+        userInt.say("Got it. I've added this task: ");
+        userInt.say("\t" + eventTask.toString());
+        userInt.say("Now you have " + dataStore.length() + " tasks in the list.");        
+    }
+
+    /**
+     *
+     * Sets a given task to completed status.
+     *
+     * @param input the user's input with the ID of the task in the list.
+     * @param dataStore the list containing the up-to-date task records.
+     * @param userInt the user interface object for logging.
+     * @param warning the exact warning to be thrown when valid index range is exceeded.
+     * @throws BotException
+     */
+    public void completeTask(String input, DataStore dataStore, Ui userInt, String warning) throws BotException {
+        int idx = Integer.parseInt(input.split(" ")[1]);
+            
+        // check if index is within appropriate range
+        if (idx < 0 || idx > dataStore.length()) {
+            throw new BotException(warning);
+        } else {
+            Task task = dataStore.get(idx-1);
+            task.setDone();
+            userInt.say("Nice! I've marked this task as done: ");
+            userInt.say("\t" + task.toString());
+        }        
+    }
+
+    /**
+     *
+     * Shuts down the bot and stops the program from running.
+     *
+     * @param userInt the user interface object for logging.
+     * @param dataStore the list containing the up-to-date task records.
+     * @param memBuff the memory buffer for reading and writing files.
+     */
+    public void shutdownBot(Ui userInt, DataStore dataStore, MemoryBuffer memBuff) {
+        // save current task log 
+        try {
+            memBuff.writeFile(dataStore);
```

Comment:
Should the writing to file only be executed on shutdown bot? What if the program gets interrupted through other means (e.g. accidentally closing the window, running out of power, ctrl + c)? Consider saving after every operation instead.